### PR TITLE
Trad 48 open orders component

### PR DESCRIPTION
**Pull request by laurvasile**

### CODE REVIEW

File: Trader.Api/Attributes/ActivityAttribute.cs [hunk 1]
```diff
@@ -47,10 +47,16 @@ public class ActivityAttribute : ActionFilterAttribute, IAsyncActionFilter
...
-                var controller = context.Controller as TradeController;
-                Activity.Message = controller.Activity.Message;
-                Activity.OrderNum = controller.Activity.OrderNum;
+                var controller = (TradeController) context.Controller;
+                Activity = _activityService.SyncFromController(Activity, controller.Activity);
             }
+
+            if (context.Controller is AccountsController)
+            {
+                var controller = (AccountsController) context.Controller;
```
Comment (sharok) [inline Trader.Api/Attributes/ActivityAttribute.cs]: Consider to refactor it. What if create interface IActivityController. And then Use this interface for controllers that you need. Also, you can delete base controller and create ActivityController and inherit this controller only when you need: TradeController/AccountController and so on. Then in ActivityAttribute you can write general code for all activty controllers. Like: context.Controller is IActivtyController then do this. Nex,t add code for specific controller, like for TradeController. In result, you will avoid duplicating.

File: Trader.Api/Controllers/TradeController.cs [hunk 2]
```diff
@@ -42,7 +45,7 @@ public class TradeController : BaseController
 
 			var dto = await _tradeService.TradeSharesAsync(tradeSharesDto);
 
-		    Activity = new ActivityDto().MapFromOrderPlacedDto(dto);
+		    Activity = _activityService.Record(new AccountTotalsDto(), dto);
```
Comment (sharok) [inline Trader.Api/Controllers/TradeController.cs]: I think it's better to create overloaded method with one parameter and handle this case in the service.

File: Trader.Api/Controllers/AccountsController.cs [hunk 3]
```diff
@@ -180,6 +180,10 @@ public class AccountsController : BaseController
                                                         await _activityService.YestardayAccountTotalsActivityAsync(),
                                                         await _brokerageAccountService.FindByIdAsync(activeAccount.ETradeAccountId)
                                                       );
+
+	        Activity = _activityService.Record(dto, new OrderPlacedDto());
```
Comment (sharok) [inline Trader.Api/Controllers/AccountsController.cs]: I think it's better to create overloaded method with one parameter and handle this case in the service.

File: Trader.Services/ActivityService.cs [hunk 4]
```diff
@@ -56,5 +56,26 @@ public ActivityService(IMongoDbContext mongoDbContext, IMapper mapper)
             var query = await _collection.Find(filter).Sort(sort).FirstOrDefaultAsync();
             return _mapper.Map<ActivityDto>(query);
         }
+
+        public ActivityDto Record(AccountTotalsDto accountTotalsDto, OrderPlacedDto orderPlacedDto)
+        {
+            var activity = new ActivityDto();
+            activity.NetAssets = accountTotalsDto.NetAssets;
```
Comment (sharok) [inline Trader.Services/ActivityService.cs]: use automapper

File: Trader.Services/ActivityService.cs [hunk 5]
```diff
@@ -56,5 +56,26 @@ public ActivityService(IMongoDbContext mongoDbContext, IMapper mapper)
...
+            activity.DayGain = accountTotalsDto.DayGain;
+            activity.Message = orderPlacedDto.Message;
+            activity.OrderNum = orderPlacedDto.OrderNum;
+
+            return activity;
+        }
+
+        public ActivityDto SyncFromController(ActivityDto attributeDto, ActivityDto controllerDto)
+        {
+            attributeDto.NetAssets = controllerDto.NetAssets;
```
Comment (sharok) [inline Trader.Services/ActivityService.cs]: use automapper

File: Trader.Api/Attributes/ActivityAttribute.cs (hunk 1 above)
Comment (laurvasile) [inline Trader.Api/Attributes/ActivityAttribute.cs]: refactored

File: Trader.Api/Controllers/TradeController.cs (hunk 2 above)
Comment (laurvasile) [inline Trader.Api/Controllers/TradeController.cs]: refactored

File: Trader.Api/Controllers/AccountsController.cs (hunk 3 above)
Comment (laurvasile) [inline Trader.Api/Controllers/AccountsController.cs]: refactored

File: Trader.Services/ActivityService.cs (hunk 4 above)
Comment (laurvasile) [inline Trader.Services/ActivityService.cs]: switched to automapper

File: Trader.Services/ActivityService.cs (hunk 5 above)
Comment (laurvasile) [inline Trader.Services/ActivityService.cs]: switched to automapper

**Pull request merged**